Core/Python/add_device_to_static_group.py

```diff
@@ -0,0 +1,173 @@
+    """
+
+    url = "https://%s/api/DeviceService/Devices?$filter=DeviceName eq \'%s\'" % (ome_ip_address, device_name)
+
+    response = requests.get(url, headers=headers, verify=False)
+    print("Getting the device ID for system with name " + device_name + "...")
+
+    if response.status_code == 200:
+        json_data = response.json()
+
+        if json_data['@odata.count'] > 1:
+            print("WARNING: We found more than one name that matched the device name: " + device_name +
+                  ". We are skipping this entry.")
+        elif json_data['@odata.count'] == 1:
+            server_id = json_data['value'][0]['Id']
+            if not isinstance(server_id, int):
+                print("The server did not return an integer ID. Something went wrong.")
+                exit(1)
+            return server_id
+        else:
+            print("WARNING: No results returned for device ID look up for name " + device_name + ". Skipping it.")
+            return 0
+    else:
+        print("Connection failed with response code " + str(response.status_code) + " while we were retrieving a "
+              "device ID from the server.")
+        exit(1)
+
+
+def add_device_to_static_group(ome_ip_address: str, ome_username: str, ome_password: str, group_name: str,
+                               device_names: list):
+    """
+    Adds a device to an existing static group
+
+    Args:
+        ome_ip_address: IP address of the OME server
+        ome_username:  Username for OME
+        ome_password: OME password
+        group_name: The group name to which you want to add servers
+        device_names: A list of device names which you want added to the group
+
+
+    """
+    """ Authenticate with OME and enumerate groups """
+    try:
+        session_url = 'https://%s/api/SessionService/Sessions' % ome_ip_address
+        group_add_device_url = "https://%s/api/GroupService/Actions/GroupService.AddMemberDevices" % ome_ip_address
+        group_url = 'https://%s/api/GroupService/Groups' % ome_ip_address
+        headers = {'content-type': 'application/json'}
+        user_details = {'UserName': ome_username,
+                        'Password': ome_password,
+                        'SessionType': 'API'}
+        group_id = 0
+
+        session_info = requests.post(session_url, verify=False,
+                                     data=json.dumps(user_details),
+                                     headers=headers)
+        if session_info.status_code == 201:
+            headers['X-Auth-Token'] = session_info.headers['X-Auth-Token']
+
+            # Resolve group ID from name
+            print("Getting the ID for group with name: " + group_name + "...")
+            response = requests.get(group_url, headers=headers, verify=False)
+            if response.status_code == 200:
+                groups = response.json()
+                found_group = False
+                for group in groups["value"]:
```

Comment:
this could use some refactoring - once the group is found there is no need to iterate through the rest of the groups - as a suggestion

may be cleaner to set group_id = -1
check if a group id matches - if so set group_id to the matching value
break out of the loop
change the boolean check to just see if it's -1 if you want to exit

Reply:
Looking into it now

Reply:
Updated.